### PR TITLE
SphereBufferGeometry: Fixed missing faces in sphere sectors

### LINKED ISSUE #6799: Sector of SphereBufferGeometry has missing faces
When rendering just a sector of a sphere, the edges are jagged due to missing faces.

    var geometry = new THREE.SphereBufferGeometry( 10, 6, 4, Math.PI / 2, Math.PI / 2, Math.PI / 4, Math.PI / 4 );

three.js r.72dev

### PR DESCRIPTION
**Pull request by WestLangley**
Fixes #6799
/ping @benaadams 

@benaadams I ran [mrdoobapproves](http://zz85.github.io/mrdoobapproves/) on the code and it changed some of the formatting. I'm not sure how you feel about that...

### CODE REVIEW

**Comment by benaadams** [comment]
:smile: :shipit: 

My editors have got confused with different coding styles for different projects, now does strange things to my code.

Need a mrdoobapproves gulp, grunt, npm plugin :wink:

**Comment by benaadams** [comment]
I am joking about the plugin btw

File: src/extras/geometries/SphereBufferGeometry.js [hunk 1]
```diff
@@ -30,10 +30,12 @@ THREE.SphereBufferGeometry = function ( radius, widthSegments, heightSegments, p
 	thetaStart = thetaStart !== undefined ? thetaStart : 0;
 	thetaLength = thetaLength !== undefined ? thetaLength : Math.PI;
 
+	var isSector = thetaStart > 0 || thetaStart + thetaLength < Math.PI;
```
Comment (mrdoob) [inline src/extras/geometries/SphereBufferGeometry.js]: I think this should be enough?

```javascript
var isSector = thetaStart + thetaLength < Math.PI;
```

Comment (WestLangley) [inline src/extras/geometries/SphereBufferGeometry.js]: Nice find! That's not quite right, either, though... We don't care about the length. We just want to know if (a) is the north pole included?, or (b) is the south pole included?. I'll try again...

Comment (mrdoob) [inline src/extras/geometries/SphereBufferGeometry.js]: Not necessarily...?

What if, for whatever reason, `thetaStart` is `Math.PI`? Think about someone using the editor and changing only the `thetaStart` value.

Comment (mrdoob) [inline src/extras/geometries/SphereBufferGeometry.js]: Actually, I confused `theta` with `phi`. Disregard the last comment 😅

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**